src/slurm_plugin/clustermgtd.py

```diff
@@ -790,16 +795,16 @@ def _handle_powering_down_nodes(self, slurm_nodes, private_ip_to_instance_map):
         Handle nodes that are powering down.
 
         Terminate instances backing the powering down node if any.
-        Reset the nodeaddr for the powering down node. Node state is not changed.
+        Do not reset the nodeaddr/nodehostname manually.
+        Nodeaddr/nodehostname will be reset automatically after power_down with cloud_reg_addrs.
+        Node state is not changed.
         """
         powering_down_nodes = []
         for node in slurm_nodes:
             if not node.is_static and node.is_nodeaddr_set() and (node.is_power() or node.is_powering_down()):
                 powering_down_nodes.append(node)
 
         if powering_down_nodes:
-            log.info("Resetting powering down nodes: %s", print_with_count(powering_down_nodes))
-            reset_nodes(nodes=[node.name for node in powering_down_nodes])
             instances_to_terminate = ClusterManager._get_backing_instance_ids(
```

Comment:
can we still retrieve the mapping here?when is the nodeaddr unset by Slurm?

Reply:
Added some comments to clarify this